igb_client/client.py

```diff
@@ -0,0 +1,127 @@
+from typing import List, Dict, Optional, Union
+
+import requests_cache
+
+from igb_client.dataclasses import JobBoard, ATSCredential, ContractCredential, Credential
+from igb_client.encrypt import AESCypher
+from igb_client.parse import parse_igb_xml_payload
+
+
+class IGBClientBase:
+    _instance = None
+    _base_url = "{base_url}/{environment_id}/{view}"
+    _credentials_storage_key = None
+
+    def __init__(self, api_key: str, environment_id: str, credentials_storage_key: str,
+                 base_url="https://api.ingoedebanen.nl/apipartner/hapi/v1"):
+        self._environment_id = environment_id
+        self._credentials_storage_key = credentials_storage_key
+        self._base_url = self._base_url.format(base_url=base_url, environment_id=environment_id)
+
+        self.session = requests_cache.CachedSession('hapi_ofccp_cache')
+        self.session.headers.update({
+            "X-IGB-Api-Key": api_key
+        })
+
+    def __new__(cls, api_key: str, environment_id: str, credentials_storage_key: str):
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+            cls._instance.__init__(api_key, environment_id, credentials_storage_key)
+        return cls._instance
+
+    def encrypt_credentials(self, credentials: Credential) -> Credential:
+        credentials.credentials = {
+            k: AESCypher(self._credentials_storage_key).encrypt(v)
+            for k, v in credentials.credentials.items()
+        }
+        return credentials
```

Comment:
Is this missing a method to decrypt those credentials?

Reply:
added helper method